lib/smart_proxy_salt_core/salt_runner.rb

```diff
@@ -0,0 +1,32 @@
+module SmartProxySaltCore
+  class SaltRunner < ForemanTasksCore::Runner::CommandRunner
+    DEFAULT_REFRESH_INTERVAL = 1
+
+    def initialize(options, suspended_action:)
+      super(options, :suspended_action => suspended_action)
+      @options = options
+    end
+
+    def start
+      command = generate_command
+      logger.debug("Running command '#{command.join(' ')}'")
+      initialize_command(*command)
+    end
+
+    def kill
+      publish_data('== TASK ABORTED BY USER ==', 'stdout')
+      publish_exit_status(1)
+      ::Process.kill('SIGTERM', @command_pid)
+    end
+
+    private
+
+    def generate_command
+      command = %w(salt)
+      command << @options['name']
+      command << 'state.template_str'
+      command << @options['script']
```

Comment:
For testing purposes, I tried to run some commands, like: "df -h" (including the surrounding template).

A error is thrown:

              stderr:
                  /bin/sh: df -h: command not found

In case a simple "df" is used, it works. Can it therefore be, that more complex script will never work?

Reply:
Seems like an escaping issue, I'll look into it. It should be definitely possible to run any script of any complexity.

Reply:
@sbernhard Could you post the rendered template somewhere?

Reply:
Works. My fault.